src/io/flutter/logging/FlutterConsoleLogManager.java

```diff
@@ -6,62 +6,246 @@
+
+  // normal, subtle, bold, dark, red
 
+  private static final ConsoleViewContentType NORMAL_CONTENT_TYPE = ConsoleViewContentType.NORMAL_OUTPUT;
+  private static final ConsoleViewContentType ERROR_CONTENT_TYPE = ConsoleViewContentType.ERROR_OUTPUT;
+  private static final ConsoleViewContentType TITLE_CONTENT_TYPE =
+    new ConsoleViewContentType("title", new SimpleTextAttributes(0, JBColor.yellow).toTextAttributes());
   private static final ConsoleViewContentType SUBTLE_CONTENT_TYPE =
     new ConsoleViewContentType("subtle", SimpleTextAttributes.GRAY_ATTRIBUTES.toTextAttributes());
+  private static final ConsoleViewContentType BOLD_CONTENT_TYPE =
+    new ConsoleViewContentType("bold", SimpleTextAttributes.REGULAR_BOLD_ATTRIBUTES.toTextAttributes());
+  private static final ConsoleViewContentType DARK_CONTENT_TYPE =
+    new ConsoleViewContentType("dark", SimpleTextAttributes.DARK_TEXT.toTextAttributes());
 
+  final private CompletableFuture<InspectorService.ObjectGroup> objectGroup;
   private static QueueProcessor<Runnable> queue;
 
   @NotNull final VmService service;
   @NotNull final ConsoleView console;
+  @NotNull final FlutterApp app;
 
-  public FlutterConsoleLogManager(@NotNull VmService service, @NotNull ConsoleView console) {
+  public FlutterConsoleLogManager(@NotNull VmService service, @NotNull ConsoleView console, @NotNull FlutterApp app) {
     this.service = service;
     this.console = console;
+    this.app = app;
+
+    if (SHOW_STRUCTURED_ERRORS) {
+      assert (app.getFlutterDebugProcess() != null);
+      objectGroup = InspectorService.createGroup(app, app.getFlutterDebugProcess(), service, "console-group");
+    }
 
     if (queue == null) {
       queue = QueueProcessor.createRunnableQueueProcessor();
     }
   }
 
   public void handleFlutterErrorEvent(@NotNull Event event) {
-    if (SHOW_STRUCTURED_ERRORS) {
-      // TODO(devoncarew): Pretty print the error (using the available console syling attributes).
-      console.print(
-        event.getExtensionData().getJson().toString() + "\n",
-        new ConsoleViewContentType("subtle", SimpleTextAttributes.GRAY_ATTRIBUTES.toTextAttributes()));
+    try {
+      final ExtensionData extensionData = event.getExtensionData();
+      final JsonObject jsonObject = extensionData.getJson().getAsJsonObject();
+      final DiagnosticsNode diagnosticsNode = new DiagnosticsNode(jsonObject, objectGroup, app, false, null);
+
+      // TODO(devoncarew): send analytics for the diagnosticsNode
+
+      if (SHOW_STRUCTURED_ERRORS) {
+        queue.add(() -> {
+          try {
+            processFlutterErrorEvent(diagnosticsNode);
+          }
+          catch (Throwable t) {
+            LOG.warn(t);
+          }
+        });
+      }
+    }
+    catch (Throwable t) {
+      LOG.warn(t);
+    }
+  }
+
+  private static final String errorSeparator =
+    "◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤";
```

Comment:
FWIW, this is the separator that on the flutter side was intended to indicate "overflow" I believe. 
I don't have a better idea for a generic separator to use. I think this is at least as reasonable as the other unicode line art used in flutter to display error boxes.

Reply:
OK, I hadn’t put that together - that is was used for overflow, I thought it was just a separator.

I do like the look of it as a separator however…

Reply:
here's the style that Flutter uses for errors by default. I agree I'm not sure which style is actually better.
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/foundation/diagnostics.dart#L557